lib/kernel/test/gen_sctp_SUITE.erl

```diff
@@ -798,6 +800,39 @@ api_connect_init(Config) when is_list(Config) ->
     ok = gen_sctp:close(Sb),
     ok.
 
+%% Test the API function connectx_init/4.
+api_connectx_init(Config) when is_list(Config) ->
+    Localhost = {127,0,0,1},
+
+    {ok,S} = gen_sctp:open(),
+    {ok,Pb} = inet:port(S),
+    try gen_sctp:connectx_init(S, Localhost, 12345, [])
+    catch error:badarg -> ok
+    end,
+    try gen_sctp:connectx_init(S, [Localhost], not_allowed_for_port, [])
+    catch error:badarg -> ok
+    end,
+    try gen_sctp:connectx_init(S, [Localhost], 12345, not_allowed_for_opts)
+    catch error:badarg -> ok
+    end,
```

Comment:
These do not test that `gen_sctp:connectx_init/4` throws a `badarg`.
Should it return something; that value is simply returned and ignored.
I think the test should verify that it *shall* throw a badarg and *not* return...

Reply:
Thanks for pointing that out, I had not considered an actual return.
Of course, the bad news is that this whole suite uses that pattern in many pre-existing test cases...

Reply:
Oops...

Reply:
How about wrapping the funtion to fail in `exit()` resulting in a pattern like this?
```erlang
try exit(gen_sctp:connectx_init(S, [Localhost], 12345, not_allowed_for_opts))
catch error:badarg -> ok
end,
```

I could make a separate PR to clean up the rest of the suite after this one is done.

Reply:
Sounds good, but I do not like using `exit(_)` for this.  Maybe this trick:
``` erlang
try error({unexpected, gen_sctp:connectx_init(S, [Localhost], 12345, not_allowed_for_opts)})
catch error : badarg -> ok
end
```

Or create a wrapper function:
``` erlang
verify_badarg(Fun) ->
    try Fun() of
        Result -> ct:fail({unexpected, Result})
    catch error : badarg -> ok
    end.

    :
    verify_badarg(
        fun () ->
            gen_sctp:connectx_init(S, [Localhost], 12345, not_allowed_for_opts)
        end),
    :
```

Reply:
In the end I made a macro since the no_return case could be useful for other tests that may not necessarily be catching `badarg`